### PR TITLE
Document and further automate setup 

### PR DESCRIPTION
**Pull request by shonfeder**
These are my notes and tiny incremental improvements from getting these set up and running locally on my machine.

See the individual commits for details.

### CODE REVIEW

File: scripts/setup-sources.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+set -euf -o pipefail
+
+latest="0.7.0"
+devl_dir="${HOME}/devl"
+
+mkdir -p "$devl_dir"
+git clone git@github.com:informalsystems/apalache.git "$devl_dir/apalache" > /dev/null
+
+for v in "0.5.1" "0.6.0" "0.7.0"
```
Comment (shonfeder) [inline scripts/setup-sources.sh:14]: Note that there is no tagged release for version 0.5.2, which is the one version specififed in the makefile. I need a pointer here about how we reconcile that.

File: scripts/setup-sources.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+do
+    echo "preparing source code for apalache version ${v}"
+    version_dir="${HOME}/devl/apalache-${v}"
+    cp -r "${devl_dir}/apalache" "$version_dir"
+    pushd "$version_dir"
+    git fetch
+    if [[ "$v" == "$latest" ]]
+    then
+        branch="origin/unstable"
```
Comment (shonfeder) [inline scripts/setup-sources.sh:23]: Is this the correct behavior for when we are testing the `latest` version, for which we have not yet cut a release? (Perhaps it should be called "unreleased"?).

File: scripts/setup-sources.sh (hunk 1 above)
Comment (konnov) [reply scripts/setup-sources.sh:14]: I don't know how we missed that. Just added `v0.5.2` for the commit that we used to build 0.5.2. It is a backport of the build system from 0.6.0 to 0.5.1.

File: scripts/setup-sources.sh (hunk 2 above)
Comment (konnov) [reply scripts/setup-sources.sh:23]: The behavior is correct. Shall we just call it `unstable`, as `latest` can be indeed ambiguous?

File: scripts/setup-sources.sh (hunk 1 above)
Comment (shonfeder) [reply scripts/setup-sources.sh]: Ah! That makes sense. I tried to run 0.5.1, only to see it fail because there was no makefile :D

**Comment by shonfeder** [comment]
Changes addressed. Thanks for the review! :)

**Pull request merged**